s4e-backend/src/test/java/pl/cyfronet/s4e/admin/scene/AdminSceneControllerTest.java

```diff
@@ -0,0 +1,323 @@
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static pl.cyfronet.s4e.TestJwtUtil.jwtBearerToken;
+
+@BasicTest
+@AutoConfigureMockMvc
+public class AdminSceneControllerTest {
+    private static final String SCHEMA_PATH_PREFIX = "classpath:schema/";
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private SchemaRepository schemaRepository;
+
+    @Autowired
+    private ProductRepository productRepository;
+
+    @Autowired
+    private SceneRepository sceneRepository;
+
+    @Autowired
+    private AppUserRepository appUserRepository;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private TestDbHelper testDbHelper;
+
+    @Autowired
+    private TestResourceHelper testResourceHelper;
+
+    private AppUser admin;
+    private AppUser user;
+
+    private Map<String, Schema> schemas;
+    private Map<String, Product> products;
+    private Map<String, List<Scene>> productScenes;
+    private long maxSceneId;
+
+    @BeforeEach
+    public void beforeEach() {
+        testDbHelper.clean();
+
+        admin = appUserRepository.save(AppUser.builder()
+                .email("admin@mail.pl")
+                .name("John")
+                .surname("Smith")
+                .password("{noop}password")
+                .enabled(true)
+                .admin(true)
+                .build());
+
+        user = appUserRepository.save(AppUser.builder()
+                .email("user@mail.pl")
+                .name("John")
+                .surname("Smith")
+                .password("{noop}password")
+                .enabled(true)
+                .build());
+
+        schemas = Stream.of("Sentinel-1.scene.v1.json", "Sentinel-1.metadata.v1.json")
+                .map(name -> {
+                    String content = new String(testResourceHelper.getAsBytes(SCHEMA_PATH_PREFIX + name));
+                    return Schema.builder()
+                            .name(name)
+                            .type(name.contains("scene") ? Schema.Type.SCENE : Schema.Type.METADATA)
+                            .content(content)
+                            .build();
+                })
+                .map(schemaRepository::save)
+                .collect(Collectors.toMap(Schema::getName, schema -> schema));
+
+        products = IntStream.range(0, 2)
+                .mapToObj(i -> SceneTestHelper.productBuilder()
```

Comment:
Could you move this builder logic to ProductFactory?

Reply:
This is already making use of SceneTestHelper and doesn't make sense to externalize.